### PR TITLE
feat: comprehensive development tooling, CI/CD, and testing improvements

### PR DESCRIPTION
**Pull request by kjanat**
## Summary
Major development infrastructure improvements including CI/CD setup, comprehensive testing suite, and modern tooling configuration.

### Development Tooling & CI/CD
- **GitHub Actions workflows** for CI, dependency review, and npm publishing
- **Biome** setup for linting and formatting with pre-commit hooks
- **Vitest** configuration for unit testing with comprehensive test suite
- **Dependabot** for automated dependency updates
- **Husky and lint-staged** for commit quality enforcement

### Testing Infrastructure  
- **100% test coverage** achieved with 22 comprehensive tests
- Tests cover all component functionality, edge cases, and type exports
- Proper TypeScript test setup with mock typing
- SVG arc path validation and color segment rendering tests

### Code Quality Improvements
- **Node.js 24** and **latest pnpm** in all workflows
- **TypeScript type safety** improvements (replaced 'any' with proper assertions)
- **Biome configuration** with .claude/ directory ignored
- **Conventional commit** message formatting

### Build & Package Improvements
- Enhanced build configuration with proper TypeScript declarations
- Multi-format builds (ESM, CJS, UMD) support
- Updated package.json with comprehensive scripts

## Test Results
- 100% statement coverage
- 100% branch coverage  
- 100% function coverage
- 100% line coverage

## Breaking Changes
None - all changes are additive improvements to development infrastructure.

## Checklist
- [x] Tests pass with 100% coverage
- [x] Linting passes (Biome + markdownlint)
- [x] Type checking passes
- [x] Build succeeds
- [x] CI workflows validated
- [x] Conventional commit format followed

## Summary by CodeRabbit

* **New Features**
  * Added automated workflows for continuous integration, publishing to npm, dependency review, and GitHub Pages deployment.
  * Introduced automated dependency updates with scheduled checks for npm packages and GitHub Actions.
  * Implemented comprehensive unit tests for the Gauge component.
  * Added configuration for pre-commit and commit message validation hooks.
  * Provided a detailed development guide and improved documentation.

* **Style**
  * Standardized code formatting and string usage across the project.
  * Improved readability in documentation and example files.

* **Chores**
  * Replaced ESLint with Biome for linting and formatting.
  * Updated and expanded development dependencies and scripts.
  * Enhanced .gitignore rules.

* **Tests**
  * Set up Vitest and Testing Library for robust test coverage.
  * Added test setup for browser API mocks.

* **Documentation**
  * Updated README for clarity and consistency.
  * Added markdown linting configuration and improved formatting.

* **Other**
  * Updated license year and holder.

### CODE REVIEW

File: biome.jsonc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      "!**/*.log",
+      "!**/*.min.js",
+      "!**/*.min.css",
+      "!pnpm-lock.yaml",
+      "!.git/**",
+      "!.husky/**",
+      "!logs/**",
+      "!.claude/**"
+    ]
+  },
```
[the remark this answers is not in the document]
Comment (kjanat) [reply biome.jsonc:25]: No, it don't: https://biomejs.dev/schemas/2.1.2/schema.json

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1 @@
+pnpm run lint-staged
```
[the remark this answers is not in the document]
Comment (kjanat) [reply .husky/pre-commit:1]: The comment you received—about missing Husky boilerplate, such as the shebang line (`#!/usr/bin/env sh`) and sourcing `_husky.sh`—was absolutely true for Husky v5 through v8, and for projects that expected cross-platform compatibility or relied on Husky’s stricter error handling in hooks.

However, **as of Husky v9+ (current major release in 2025), the requirement to include those specific lines in your `.husky/pre-commit` and similar hook files has changed**:

- **For Husky v8 and older**: The shebang and sourcing line were required. Without them, the hook might not run, especially across different shells or platforms, or Git might not treat the file as executable, leading to the "hook will be skipped" issue.
- **For Husky v9 and later**: There was a migration to simplify the hook scripts. Husky now considers these lines redundant and actually asks you to remove them starting from version 9.1.1. If you keep them, you may even see a message warning that they’ll fail in v10.0.0. Husky uses its own mechanism for hook invocation and error handling now, so the traditional boilerplate is no longer mandatory or recommended for new projects[^1][^2].

**Concrete example for v9+:**

- Your `.husky/pre-commit` might only contain a single command, like:

  ```sh
  npm run lint-staged
  ```

  or whatever action you want, and modern Husky setups handle the rest.

**Summary:**

- If you are using Husky v9 or later (which you almost certainly are in 2025), *the strict requirement for the old boilerplate is obsolete*. You can omit the shebang and the `. "$(dirname -- "$0")/_/husky.sh"` line in your hook files[^1][^2].
- If you have legacy contributors or CI pipelines still running older Husky (v8 or below), or if your repo must support old setups, keeping the boilerplate would avoid issues.
- For pure v9+ projects: **Just remove the lines, as advised in recent Husky releases.**

> **In conclusion:**  
> The PR comment is outdated for Husky v9+. The modern guidance is to remove those boilerplate lines to prevent future breakage and unnecessary complexity[^1][^2].

[^1]: https://github.com/typicode/husky/issues/1476
[^2]: https://dev.classmethod.jp/articles/husky-changed-v8-to-v9-migration/
[^3]: https://github.com/typicode/husky
[^4]: https://oliviac.dev/blog/set_up_pre_commit_hook_husky_lint_staged/
[^5]: https://dev.to/zhangzewei/pre-commit-with-husky-lint-staged-2kcm
[^6]: https://github.com/react-boilerplate/react-boilerplate/issues/1895
[^7]: https://dev.to/mahmudulhsn/install-husky-in-your-project-for-proper-commit-lint-with-pre-commit-hooks-25b2
[^8]: https://stackoverflow.com/questions/67198286/how-to-make-shell-agnostic-a-husky-hook
[^9]: https://dev.to/krzysztofkaczy9/do-you-really-need-husky-247b
[^10]: https://www.youtube.com/watch?v=j3bmgO_jtXo
[^11]: https://typicode.github.io/husky/
[^12]: https://typicode.github.io/husky/how-to.html
[^13]: https://blog.logrocket.com/build-robust-react-app-husky-pre-commit-hooks-github-actions/
[^14]: https://www.git-tower.com/blog/git-hooks-husky
[^15]: https://blog.knappi.org/0010-why-you-dont-need-husky/
[^16]: https://stackoverflow.com/questions/8598639/why-is-my-git-pre-commit-hook-not-executable-by-default
[^17]: https://stackoverflow.com/questions/66631444/how-do-i-resolve-husky-husky-sh-no-such-file-or-directory-error
[^18]: http://www.seanerikoconnor.freeservers.com/ComputerScience/DevelopmentEnvironment/macLinuxComputerSetupForDevelopers.html
[^19]: https://github.com/typicode/husky/issues/869
[^20]: https://snyk.io/advisor/npm-packages-list

File: .github/workflows/pages.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '24'
+          cache: 'pnpm'
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v5
+
```
[the remark this answers is not in the document]
Comment (kjanat) [reply .github/workflows/pages.yml:36]: no thx, will cause too many dependabot pr's which are mostly irrelevant

**Comment by kjanat** [comment]
This pull request introduces comprehensive updates to the project's configuration, workflows, and documentation, focusing on automation, code quality, and developer experience improvements. Key changes include the addition of GitHub Actions workflows for CI/CD, dependency management, and publishing, as well as updates to linting and formatting configurations. The project's development guide has been expanded, and minor adjustments have been made to the codebase and documentation for consistency.

### GitHub Actions Workflows

* [`.github/workflows/ci.yml`](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR1-R118): Added a CI pipeline to automate linting, type checking, building, testing, and security scanning for `master` and `develop` branches.
* [`.github/workflows/dependency-review.yml`](diffhunk://#diff-7cdd3ccec44c8ba176bdc3b9ef54c3f56aa210a1a4e2bb5f79d87b1e50314a18R1-R56): Introduced a workflow to review dependency changes, check for vulnerabilities, and ensure license compliance in pull requests.
* [`.github/workflows/pages.yml`](diffhunk://#diff-267452bb53d388a7b14551d9fc53a797edc124dc9085ecc79f5339277a1e53b5R1-R57): Added a workflow to deploy examples to GitHub Pages on pushes to `master`.
* [`.github/workflows/publish.yml`](diffhunk://#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R1-R57): Implemented a workflow for publishing the library to npm on GitHub releases, with support for manual version input.

### Code Quality and Automation Enhancements

* [`.github/dependabot.yml`](diffhunk://#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28R1-R37): Configured Dependabot to check for weekly updates to npm dependencies and GitHub Actions, with separate groups for development and production dependencies.
* [`.husky/commit-msg`](diffhunk://#diff-292bac03df29207ee254ba91cfc30951b10c50c0a9fff3a0768778cebe5f4c4aR1-R79): Added a commit message validator enforcing conventional commit formats and rejecting invalid messages.
* [`.husky/pre-commit`](diffhunk://#diff-d2bc4bbf14eadc292d84e0ef5f7a93115c23557f533809fc80b896934291529dR1): Configured a pre-commit hook to run lint-staged for code quality checks.
* [`.lintstagedrc.json`](diffhunk://#diff-3192e47cadb571a6fe5cc66fe1a672718e75903e3605f67b04f31ed3cb4493baR1-R4): Added lint-staged configuration to automatically format and lint staged files using Biome and markdownlint.
* [`.biome.jsonc`](diffhunk://#diff-955d6dd0b9089fa951f33ccabbab4597fe019091b7bb81e166df2b34dea4a906R1-R81): Defined Biome configuration for linting, formatting, and project-specific rules.

### Documentation Updates

* [`DEVELOPMENT.md`](diffhunk://#diff-fc2a310fcfedfefb0046def697f932def80cbb1e78c689bc0af3c8eab3e33eceR1-R247): Added a detailed development guide covering setup, workflows, scripts, CI/CD pipelines, and coding standards.
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L35-R42): Updated code examples for better readability and consistency. Minor adjustments to formatting and language. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L35-R42) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L68-R67) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R162)

### Minor Code Adjustments

* [`examples/App.css`](diffhunk://#diff-8bfda526e23a847fc01621421fb160ff8d50f2dc398a11fcd8e9aaf9f8003fc8L3-R5): Reformatted font-family property for consistency.
* [`examples/App.tsx`](diffhunk://#diff-fc99504323bd60153a3f1b922ebd084294a6939b42d1045554101a394cdfee0dL1-R4): Adjusted import statements to use double quotes for consistency with Biome configuration.

### Licensing

* [`LICENSE`](diffhunk://#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7L3-R3): Updated copyright information to reflect the new copyright holder.
